service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/transport/amqps/AmqpFeedbackReceivedHandler.java

```diff
@@ -154,49 +193,34 @@ public void onDelivery(Event event)
     }
 
     @Override
-    public void onConnectionInit(Event event)
-    {
-        // Codes_SRS_SERVICE_SDK_JAVA_AMQPFEEDBACKRECEIVEDHANDLER_12_011: [The event handler shall set the host name on the connection]
-        Connection conn = event.getConnection();
-        conn.setHostname(hostName);
-
-        // Every session or link could have their own handler(s) if we
-        // wanted simply by adding the handler to the given session
-        // or link
-
-        // Codes_SRS_SERVICE_SDK_JAVA_AMQPFEEDBACKRECEIVEDHANDLER_12_012: [The event handler shall create a Session (Proton) object from the connection]
-        Session ssn = conn.session();
-
-        // If a link doesn't have an event handler, the events go to
-        // its parent session. If the session doesn't have a handler
-        // the events go to its parent connection. If the connection
-        // doesn't have a handler, the events go to the reactor.
-
-        // Codes_SRS_SERVICE_SDK_JAVA_AMQPFEEDBACKRECEIVEDHANDLER_12_013: [The event handler shall create a Receiver (Proton) object and set the protocol tag on it to a predefined constant]
-        // Codes_SRS_SERVICE_SDK_JAVA_AMQPFEEDBACKRECEIVEDHANDLER_15_017: [The Receiver object shall have the properties set to service client version identifier.]
-        Map<Symbol, Object> properties = new HashMap<>();
-        properties.put(Symbol.getSymbol(TransportUtils.versionIdentifierKey), TransportUtils.USER_AGENT_STRING);
-        feedbackReceiverLink = ssn.receiver(RECEIVE_TAG);
-        feedbackReceiverLink.setProperties(properties);
-
-        // Codes_SRS_SERVICE_SDK_JAVA_AMQPFEEDBACKRECEIVEDHANDLER_12_014: [The event handler shall open the Connection, the Session and the Receiver object]
-        log.debug("Opening connection, session and link for amqp feedback receiver");
-        conn.open();
-        ssn.open();
-        feedbackReceiverLink.open();
```

Comment:
Session and link opening is now deferred until the cbs authentication completes successfully